File: 0038-count-and-say/0038-count-and-say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        if n == 1:
            return "1"
        
        # Start with the base case
        result = "1"
        
        # Generate the sequence up to the nth term
        for _ in range(1, n):
            current = ""
            count = 1
            char = result[0]
            
            for i in range(1, len(result)):
                if result[i] == char:
                    count += 1
                else:
                    current += str(count) + char
                    char = result[i]
                    count = 1
            
            current += str(count) + char
            result = current
        
        return result

File: 0038-count-and-say/0038-count-and-say.java
class Solution {
    public String countAndSay(int n) {
        if (n == 1) return "1";
        
        String prev = countAndSay(n - 1);
        StringBuilder result = new StringBuilder();
        
        int count = 1;
        char say = prev.charAt(0);
        
        for (int i = 1; i < prev.length(); i++) {
            if (prev.charAt(i) == say) {
                count++;
            } else {
                result.append(count).append(say);
                count = 1;
                say = prev.charAt(i);
            }
        }
        
        result.append(count).append(say);
        
        return result.toString();
    }
}
